F411re compiling issue

```
C:\nf-interpreter\targets\CMSIS-OS\ChibiOS\nf-overlay\os\hal\ports\STM32\LLD\FLASHv2\flash_lld.c: In function 'flash_lld_getSector':
C:\nf-interpreter\targets\CMSIS-OS\ChibiOS\nf-overlay\os\hal\ports\STM32\LLD\FLASHv2\flash_lld.c:276:22: error: 'ADDR_FLASH_SECTOR_8' undeclared (first use in this function)
   else if((address < ADDR_FLASH_SECTOR_8) && (address >= ADDR_FLASH_SECTOR_7))
                      ^~~~~~~~~~~~~~~~~~~
```
there is no in ADDR_FLASH_SECTOR_8 defined in flash_lld.h

```
#if defined(STM32F401xE) || defined(STM32F411xE) || defined(STM32F446xx)

#define FLASH_SECTOR_0     ((uint32_t)0U) /*!< Sector Number 0   */
#define FLASH_SECTOR_1     ((uint32_t)1U) /*!< Sector Number 1   */
#define FLASH_SECTOR_2     ((uint32_t)2U) /*!< Sector Number 2   */
#define FLASH_SECTOR_3     ((uint32_t)3U) /*!< Sector Number 3   */
#define FLASH_SECTOR_4     ((uint32_t)4U) /*!< Sector Number 4   */
#define FLASH_SECTOR_5     ((uint32_t)5U) /*!< Sector Number 5   */
#define FLASH_SECTOR_6     ((uint32_t)6U) /*!< Sector Number 6   */
#define FLASH_SECTOR_7     ((uint32_t)7U) /*!< Sector Number 7   */

#define ADDR_FLASH_SECTOR_0     ((uint32_t)0x08000000) /* Base @ of Sector 0, 16 Kbyte */
#define ADDR_FLASH_SECTOR_1     ((uint32_t)0x08004000) /* Base @ of Sector 1, 16 Kbyte */
#define ADDR_FLASH_SECTOR_2     ((uint32_t)0x08008000) /* Base @ of Sector 2, 16 Kbyte */
#define ADDR_FLASH_SECTOR_3     ((uint32_t)0x0800C000) /* Base @ of Sector 3, 16 Kbyte */
#define ADDR_FLASH_SECTOR_4     ((uint32_t)0x08010000) /* Base @ of Sector 4, 64 Kbyte */
#define ADDR_FLASH_SECTOR_5     ((uint32_t)0x08020000) /* Base @ of Sector 5, 128 Kbyte */
#define ADDR_FLASH_SECTOR_6     ((uint32_t)0x08040000) /* Base @ of Sector 6, 128 Kbyte */
#define ADDR_FLASH_SECTOR_7     ((uint32_t)0x08060000) /* Base @ of Sector 7, 128 Kbyte */


#endif /* STM32F401xE || STM32F411xE || STM32F446xx */
```
so I added 
#define ADDR_FLASH_SECTOR_8     ((uint32_t)0x08080000) /* Base @ of Sector 8, 128 Kbyte */
into .h file.

Current compiling error log:
```
c:/gcc-arm-none-eabi-6-2017-q1-update-win32/bin/../lib/gcc/arm-none-eabi/6.3.1/../../../../arm-none-eabi/bin/ld.exe: nanoBooter.elf section `.rodata' will not fit in region `flash'
c:/gcc-arm-none-eabi-6-2017-q1-update-win32/bin/../lib/gcc/arm-none-eabi/6.3.1/../../../../arm-none-eabi/bin/ld.exe: region `flash' overflowed by 1468 bytes
collect2.exe: error: ld returned 1 exit status
```
It's a long time issue.

--- a/targets/CMSIS-OS/ChibiOS/nf-overlay/os/hal/ports/STM32/LLD/FLASHv2/flash_lld.c
+++ b/targets/CMSIS-OS/ChibiOS/nf-overlay/os/hal/ports/STM32/LLD/FLASHv2/flash_lld.c
@@ -272,8 +272,16 @@ uint8_t flash_lld_getSector(uint32_t address)
     sector = FLASH_SECTOR_6;
   }
 #endif
-#if defined(FLASH_SECTOR_7)
+// the next one is for devices that have FLASH sector 8 defined (see next one for devices WITHOUT sector 8)
+#if defined(FLASH_SECTOR_7) && defined(FLASH_SECTOR_8)
   else if((address < ADDR_FLASH_SECTOR_8) && (address >= ADDR_FLASH_SECTOR_7))
+  {
+    sector = FLASH_SECTOR_7;
+  }
+#endif
+// this next one needs to be here for devices that don't have FLASH sector 8
+#if defined(FLASH_SECTOR_7) && !defined(FLASH_SECTOR_8)
+  else if(address >= ADDR_FLASH_SECTOR_7)
   {
     sector = FLASH_SECTOR_7;
   }

